arangod/RocksDBEngine/RocksDBVPackIndex.cpp

```diff
@@ -2858,4 +2859,155 @@ void RocksDBVPackIndex::warmupInternal(transaction::Methods* trx) {
+                             RocksDBVPackStreamOptions options)
+      : _index(index),
+        _options(std::move(options)),
+        _bounds(
+            isUnique
+                ? RocksDBKeyBounds::UniqueVPackIndex(index->objectId(), false)
+                : RocksDBKeyBounds::VPackIndex(index->objectId(), false)) {
+    _end = _bounds.end();
+    RocksDBTransactionMethods* mthds =
+        RocksDBTransactionState::toMethods(trx, index->collection().id());
+    _iterator = mthds->NewIterator(index->columnFamily(), [&](auto& opts) {
+      TRI_ASSERT(opts.prefix_same_as_start);
+      opts.iterate_upper_bound = &_end;
+    });
+    _iterator->Seek(_bounds.start());
+  }
+
+  bool position(std::span<VPackSlice> span) const override {
+    if (!_iterator->Valid()) {
+      return false;
+    }
+
+    TRI_ASSERT(RocksDBKey::objectId(_iterator->key()) == _index->objectId())
+        << RocksDBKey::objectId(_iterator->key())
+        << " actual = " << _index->objectId();
+
+    auto keySlice = RocksDBKey::indexedVPack(_iterator->key());
+    storeKey(span, keySlice);
+    return true;
+  }
+
+  void loadKey(std::span<VPackSlice> span) {
+    _builder.clear();
+    _builder.openArray();
+    for (auto k : span) {
+      _builder.add(k);
+    }
+    _builder.close();
+  }
+
+  void storeKey(std::span<VPackSlice> into, VPackSlice keySlice) const {
+    std::size_t idx = 0;
+    for (auto k : VPackArrayIterator(keySlice)) {
+      if (idx >= _options.keyPrefixSize) {
+        break;
+      }
+      into[idx] = k;
+    }
+  }
+
+  bool seek(std::span<VPackSlice> span) override {
+    loadKey(span);
+    RocksDBKey rkey;
+    rkey.constructUniqueVPackIndexValue(_index->objectId(), _builder.slice());
+    _iterator->Seek(rkey.string());
+    return position(span);
+  }
+
+  LocalDocumentId load(std::span<VPackSlice> projections) const override {
+    // TODO implement projections
+    if constexpr (isUnique) {
+      return RocksDBValue::documentId(_iterator->key());
+    } else {
+      return RocksDBKey::indexDocumentId(_iterator->key());
+    }
+  }
+
+  bool next(std::span<VPackSlice> key, LocalDocumentId& docId,
+            std::span<VPackSlice> projections) override {
+    _iterator->Next();
+    if (!_iterator->Valid()) {
+      return false;
+    }
+
+    storeKey(key, RocksDBKey::indexedVPack(_iterator->key()));
+    docId = load(projections);
+    return true;
+  }
+
+  void cacheCurrentKey(std::span<VPackSlice> cache) override {
+    _cache = VPackString{RocksDBKey::indexedVPack(_iterator->key())};
+    storeKey(cache, _cache);
+  }
+
+  bool reset(std::span<VPackSlice> span) override {
+    _iterator->Seek(_bounds.start());
+    return position(span);
+  }
+};
+
+}  // namespace
+
+std::unique_ptr<AqlIndexStreamIterator> RocksDBVPackIndex::streamForCondition(
+    transaction::Methods* trx, IndexStreamOptions const& opts) {
+  if (!supportsStreamInterface(opts)) {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_BAD_PARAMETER);
```

Comment:
Let's use TRI_ERROR_INTERNAL here, because it is not really an error the user could fix. We should also use `THROW_ARANGO_EXCEPTION_MESSAGE(code, msg)` with a more descriptive error message. Only getting "Bad parameter" will otherwise be hard to debug if this is ever called.
```suggestion
    THROW_ARANGO_EXCEPTION(TRI_ERROR_BAD_PARAMETER);
```